File: scripts/inject-all.py
import os

files = [
    ('src/pages/BlogPostMSKIAM.jsx', 'aws-msk-iam-auth'),
    ('src/pages/BlogPostMentoring.jsx', 'mentoring-interns-in-tech'),
    ('src/pages/BlogPostRateLimit.jsx', 'rate-limiting-algorithms'),
    ('src/pages/BlogPostSigstore.jsx', 'sigstore-software-supply-chain'),
    ('src/pages/BlogPostCursor.jsx', 'cursor-tdd-refactoring'),
    ('src/pages/BlogPostGCP.jsx', 'create-mute-rule-gcp-scc')
]

for filepath, slug in files:
    if not os.path.exists(filepath): continue
    
    with open(filepath, 'r') as f:
        content = f.read()
        
    if 'import ReadProgress' not in content:
        # 1. Imports
        content = content.replace(
            "import SEO from '../components/SEO';",
            "import SEO from '../components/SEO';\nimport ReadProgress from '../components/ReadProgress';\nimport ReactionWidget from '../components/ReactionWidget';\nimport BlogComments from '../components/BlogComments';"
        )
        
        # 2. ReadProgress
        content = content.replace(
            '<div className="min-h-screen bg-white pt-20">',
            '<div className="min-h-screen bg-white pt-20">\n            <ReadProgress />'
        )
        
        # 3. Widgets at bottom
        content = content.replace(
            '</article>',
            f'    <ReactionWidget slug="{slug}" />\n                <BlogComments />\n            </article>'
        )
        
        with open(filepath, 'w') as f:
            f.write(content)
        
        print(f"Updated {filepath}")

File: scripts/inject-all.js
const fs = require('fs');

const files = [
  { path: 'src/pages/BlogPostMSKIAM.jsx', slug: 'aws-msk-iam-auth' },
  { path: 'src/pages/BlogPostMentoring.jsx', slug: 'mentoring-interns-in-tech' },
  { path: 'src/pages/BlogPostRateLimit.jsx', slug: 'rate-limiting-algorithms' },
  { path: 'src/pages/BlogPostSigstore.jsx', slug: 'sigstore-software-supply-chain' },
  { path: 'src/pages/BlogPostCursor.jsx', slug: 'cursor-tdd-refactoring' },
  { path: 'src/pages/BlogPostGCP.jsx', slug: 'create-mute-rule-gcp-scc' }
];

files.forEach(f => {
  let content = fs.readFileSync(f.path, 'utf8');
  
  if (!content.includes('import ReadProgress')) {
    content = content.replace(
      "import SEO from '../components/SEO';", 
      "import SEO from '../components/SEO';\nimport ReadProgress from '../components/ReadProgress';\nimport ReactionWidget from '../components/ReactionWidget';\nimport BlogComments from '../components/BlogComments';"
    );
    
    content = content.replace(
      '<div className="min-h-screen bg-white pt-20">', 
      '<div className="min-h-screen bg-white pt-20">\n            <ReadProgress />'
    );
    
    content = content.replace(
      '</article>', 
      `    <ReactionWidget slug="${f.slug}" />\n                <BlogComments />\n            </article>`
    );
    
    fs.writeFileSync(f.path, content);
    console.log('✅ Updated ' + f.path);
  }
});
